### PR TITLE
DockerTaskAPIService and process utils

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**

### CODE REVIEW

File: python/golem_task_api/testutils/dockertaskapiservice.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    def running(self) -> bool:
+        try:
+            self._container.reload()
+        except (AttributeError, docker.errors.APIError):
+            return False
+        return self._container.status not in ['exited', 'error']
+
+    async def wait_until_shutdown_complete(self) -> None:
+        print(f'Shutting down container with status: {self._container.status}')
```
Comment (maaktweluit) [inline python/golem_task_api/testutils/dockertaskapiservice.py]: afair .status is a cached property, to read the most recent values you need to call another FN, can not find it so quickly but will look more

File: python/golem_task_api/testutils/dockertaskapiservice.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        self._work_dir = work_dir
+        self._vm_name = vm_name
+        self._container = None
+
+    async def start(
+            self,
+            command: str,
+            port: int
+    ) -> Tuple[str, int]:
+        self._container = run_docker_container(
```
Comment (Wiezzel) [inline python/golem_task_api/testutils/dockertaskapiservice.py]: The problem with this approach is that it uses synchronous calls which kind of defeats the purpose of making `start` method async. Maybe we could use [aiodocker](https://github.com/aio-libs/aiodocker) here?

File: python/golem_task_api/testutils/dockertaskapiservice.py (hunk 1 above)
Comment (Wiezzel) [reply python/golem_task_api/testutils/dockertaskapiservice.py]: It's `reload`. It is already being called in `running()`.

File: python/golem_task_api/testutils/dockertaskapiservice.py (hunk 2 above)
Comment (mfranciszkiewicz) [reply python/golem_task_api/testutils/dockertaskapiservice.py]: How did the original DockerTaskAPIService implementation got accepted then? This is based off of the original one and conforms to the service interface.

I don't think that aiodocker (https://github.com/golemfactory/task-api/pull/60 + working branch in tutorialapp repo) brings any added value to testutils. It requires changes to the base service interface and lacks in functionality.

**Pull request merged**